src/main/java/me/gnat008/perworldinventory/groups/Group.java

```diff
@@ -56,9 +60,41 @@ public Group(String name, List<String> worlds, GameMode gameMode) {
      * @param configured If the group is defined in the worlds.yml file.
      */
     public Group(String name, List<String> worlds, GameMode gameMode, boolean configured) {
+        this(name,worlds,null,null,null,null,gameMode,configured);
+    }
+
+
+    /**
+     * Constructor.
+     *
+     * @param name The name of the group.
+     * @param worlds A list of world names in this group.
+     * @param gameMode The default {@link GameMode} for this group.
+     * @param useLastWorld A {@link List} of teleport causes that should map to the last world in group
+     * @param usePosInWorld A {@link List} of teleport causes that should map to the last position in world
+     */
+    public Group(String name, List<String> worlds, List<String> useLastWorld, List<String> useLastPosInGroup, List<String> useLastPosInWorld, String defaultWorld, GameMode gameMode) {
+        this(name,worlds,useLastWorld,useLastPosInGroup,useLastPosInWorld,defaultWorld,gameMode,false);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param name The name of the group.
+     * @param worlds A list of world names in this group.
+     * @param gameMode The default {@link GameMode} for this group.
+     * @param useLastWorld A {@link List} of teleport causes that should map to the last world in group
+     * @param usePosInWorld A {@link List} of teleport causes that should map to the last position in world
+     * @param configured If the group is defined in the worlds.yml file.
+     */
+    public Group(String name, List<String> worlds, List<String> useLastWorld, List<String> useLastPosInGroup, List<String> useLastPosInWorld, String defaultWorld, GameMode gameMode, boolean configured) {
```

Comment:
The new `List<String>` fields, absolutely, forcibly, _must_ be enums. Nothing in the code suggests otherwise that this is only a set of predefined entries, and I've actually only started to understand this after having looked at the worlds.yml. 
All those entries should be of type `PlayerTeleportEvent.TeleportCause`.